src/main/java/org/example/services/Services.java

```diff
@@ -25,59 +24,51 @@ public static boolean isPerfectNumber(int artifactCode) {
                 sum += i;
             }
         }
-        if (sum == artifactCode) {
+
+        if (artifactCode == 0) {
+            return isPerfect;
+        } else if (sum == artifactCode) {
             isPerfect = true;
         }
```

Comment:
The logic needs refactor to be simpler and more readable.

Reply:
Solved